### ISSUE TITLE
Chapter05 - Wrong output and incomplete example

### ISSUE THREAD

**Issue by IniterWorker** [PacktPublishing/Hands-On-Functional-Programming-in-RUST#4]
# Chapter 05
- The software application doesn't fit with the requirement and doesn't work.

# Behavior

Inputs :

```
1
1
2
```

Outputs : 
```
[ ]   Carriage at floor 1                                                                                                            
[ ]   Location          -123765.633135                                                                                               
[ ]   Velocity          -1557.509385                                                                                                 
[ ]   Acceleration      -9.800000                                                                                                    
[ ]   Force [up-down]   0.000000                                                                                                     
[ ]                                                                                                                                  
[ ]                                                                                                                                  
[X]                                                                                                                                                                               
```

> Overall, the deliverables are supported by fairly maintainable code, and the project is going good.

- I'm not sure about "going good." I used the repository to do my tests.

- Another thing that is the absence of input text file reference in this Chapter05.
-- Maybe the test process is important too! 

- I have a question about it. Why did you remove the tests folder? 
-- The end-to-end test was basic, but who knows.

Best regards,

Walter

**Comment by andrew4328** [comment]
Could you clarify which executable you are running to input this and get this error. I assume this is a simulate_trip?

Also, sorry for the poor debugging code, this could be much better.

**Comment by andrew-packtpub** [comment]
Thanks for the report, this is awefully confusing. The C motor controllers are left as stubs. I'm not sure if this is mentioned in the book or not, but without working motor controller code, the carriage just falls due to gravity. Since this is undocumented in the book and probably just bad practice, I will be providing a patch with working motor controllers over the next few days. Should get this particular issue resolved then.

**Comment by andrew-packtpub** [comment]
Check current master, it should fix the issue: 4624de4e42506f5b2b94005c7f624d60f9f3bb1f

**Comment by deepesh-packt** [comment]
Thanks!
Please feel free to reopen if the issue still persists.
